clippy_lints/src/copies.rs

```diff
@@ -185,7 +185,9 @@ fn lint_match_arms(cx: &LateContext<'_, '_>, expr: &Expr) {
         };
 
         let indexed_arms: Vec<(usize, &Arm)> = arms.iter().enumerate().collect();
-        if let Some((&(_, i), &(_, j))) = search_same(&indexed_arms, hash, eq) {
+        for match_expr in search_same_list(&indexed_arms, hash, eq) {
```

Comment:
You could also destructure in the for loop to mirror the previous implementation & remove the next line.

```suggestion
for (&(_, i), &(_, j)) in search_same_list(&indexed_arms, hash, eq) {
```